libkbfs/journal_md_ops.go

```diff
@@ -158,7 +178,21 @@ func (j journalMDOps) getRangeFromJournal(
 		return nil, nil
 	}
 
-	head := ibrmds[len(ibrmds)-1]
+	headIndex := len(ibrmds) - 1
```

Comment:
Similar to the above, you can save some work by doing the filtering in `mdJournal`, and having it take a `bid BranchID` in `getRange`. The special case is if `bid` is either `NullBranchID` or `Pending...` and `mdJournal.branchID` is `Pending...`, and you can find the cutoff point and restrict the range appropriately.